prow/gerrit/client/client.go

```diff
@@ -685,7 +685,7 @@ func (h *gerritInstanceHandler) queryChangesForProjectWithoutMetrics(log logrus.
 
 		if changes == nil || len(*changes) == 0 {
 			log.Info("No more changes")
-			return pending, nil
+			goto return_result
```

Comment:
https://xkcd.com/292/

Seriously, I think this use of goto is clean enough. Could label the outer for-loop and use "break outer" too.

Reply:
> Could label the outer for-loop and use "break outer" too.

True. But I prefer the consistency and simplicity of using the goto in both places (instead of being clever and saying in one of the places "hey I don't need to use a goto here because we're in the outer loop").

Reply:
Hmm, sorry, I misread your comment. You were suggesting the use of 2 labels, I think? But I think the current approach is simpler.

Reply:
Sorry, what I meant was to place a label above the outermost 'for' loop. True, the break outside the inner loop wouldn't need the label, but it is allowed.

Reply:
> Sorry, what I meant was to place a label above the outermost 'for' loop

A label above the outer loop would let statements jump to that location to re-execute the outer loop -- not sure why we would want that here. If you want you could do the edit and push up a branch to your fork to demonstrate what you mean.